src/components/Button/Button.scss

```diff
@@ -349,11 +349,16 @@
     background-color: transparent;
   }
 
-  .ms-Icon {
+  .ms-Icon--m {
     font-size: $ms-icon-size-m;
     padding: 8px;
   }
 
+  .ms-Icon--s {
```

Comment:
Sounds good. I was just following the approach done for the medium size. Switched to BEM'ish modifier class.